React capstone project - Metrics webapp

denisdiaconu opened the pull request:
This React capstone project is about building a mobile web application to check a list of metrics.

I did the following:

-   Use React documentation.
-   Use React components.
-   Use React props.
-    Use React Router.
-    Connect React and Redux.
-    Handle events in a React app.
-    Write integration tests with a React testing library.
-    Use styles in a React app.
-    Use React life cycle methods.
-    Apply React best practices and language style guides in code.
-    Use store, actions and reducers in React.

denisdiaconu committed:
Initial commit

denisdiaconu committed:
Add redux and routes

denisdiaconu committed:
Add action to fetch anime from the api

denisdiaconu committed:
Add entire logic of the project

denisdiaconu committed:
Add bootstrap

denisdiaconu committed:
Add style for navbar and details

denisdiaconu committed:
Add final design for Details page

denisdiaconu committed:
Add design for home

denisdiaconu committed:
Fix linters errors

denisdiaconu committed:
Add fix for details

denisdiaconu committed:
Add testing for actions

denisdiaconu committed:
Add image for the readme

denisdiaconu committed:
Add deploy

denisdiaconu committed:
Fix display problem

denisdiaconu committed:
Add base url for route

denisdiaconu committed:
Add script in package

denisdiaconu committed:
New github page

denisdiaconu committed:
Add new script

denisdiaconu committed:
Update README.md

denisdiaconu committed:
Update README.md

oloomoses commented on src/components/Details.js:
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable */
+import './style/Details.css';
```
- [x] Kindly enable the linter in this file. Linter enforces quality code standards and therefore should not be dissabled.

oloomoses commented on src/api/api.js:
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable */
+import axios from 'axios';
```
- [x] Kindly enable the linter in this file. Linter enforces quality code standards and therefore should not be dissabled.

oloomoses commented on src/redux/cofigureStore.js:
```diff
@@ -0,0 +1,15 @@
+  animeReducer,
+});
+
+const store = createStore(
+  reducer,
+  applyMiddleware(thunk, logger),
+);
```
- [x] Please do not log redux data on the browser console as it may cause site security issues. Kindly clear the logged data displayed on the browser console.

oloomoses commented on src/components/Header.js:
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable */ 
+import React from 'react';
```
- [x] Kindly enable the linter in this file. Linter enforces quality code standards and therefore should not be disabled.

oloomoses commented on public/index.html:
```diff
@@ -0,0 +1,58 @@
+    <!--
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <!--
```
- [x] This manifest link is generating an error in the browser console. Kindly remove it.

oloomoses commented on public/index.html:
```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
```
- [x] This icon link is generating an error in the browser console. Kindly remove it.

oloomoses reviewed (changes requested):
Hi @denisdiaconu,

Great video presentation. Well done on the UI and the readme.
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] I expected to see more linter checks on the pull request. Kindly check that to ensure that linters is enabled on all files and in GitHub settings.
- [x] Kindly add more tests for the app. You should add unit tests for the components and pure the functions, have mock tests for the API requests.
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Slack: Oloo Moses
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

denisdiaconu committed:
Add all the changes

denisdiaconu commented:
Hello,

I did the following changes:

-  I removed all the /* eslint-disable */ 
-  I removed the logger from the store
-  I removed the manifest
-  I removed the icon link

Georjane reviewed (approved):
## Status: PR Approved 💯 🥇 
#### Hello @denisdiaconu, 

Your project is complete!  You have met all requirements for this project. It is a thumbs up for me. 👍🏽 You can go ahead and merge this PR. 
Congratulations! 🎉 👏👏

_Good luck with your next project and Happy coding!_ 💻
